python/pyspark/sql/connect/plan.py

```diff
@@ -721,8 +721,17 @@ def plan(self, session: "SparkConnectClient") -> proto.Relation:
                     rel.join.using_columns.append(self.on)
                 else:
                     rel.join.join_condition.CopyFrom(self.to_attr_or_expression(self.on, session))
-            else:
-                rel.join.using_columns.extend(self.on)
+            elif len(self.on) > 0:
```

Comment:
Should we throw an exception if the list (or string) is empty?

Reply:
I actually assumed Catalyst will complain if it hits illegal case. 

Do you think client side should be more aggressive to check things before passing to server?

Reply:
After check the Scala Dataset, I believe that does not do such validation so I assume Dataset also choose to leave Calalyst to handle such case.